### Investigate 14999 null dates in item attempt dates

In [7]:
%run '../lib/libraries.ipynb'

dython              0.6.1
tinydb              4.2.0


You should consider upgrading via the '/Users/duane.hinkley/PycharmProjects/c772-capstone-project/venv/bin/python -m pip install --upgrade pip' command.
You should consider upgrading via the '/Users/duane.hinkley/PycharmProjects/c772-capstone-project/venv/bin/python -m pip install --upgrade pip' command.


In [11]:
dfFlt  = load_df('dfFlt')
types = get_var_types()

###### Create Dataframe of null observations

In [9]:
nullDf = dfFlt.filter(F.col('item_attempt_start_datetime_utc').isNull())
nullDf.count()

14999

Dataframe contains the 14,999 null item_attempt_start_datetime_utc


###### Display distinct values

In [10]:
distinct_val(nullDf)

0
assignment_max_attempts                1
assignment_late_submission         False
assignment_attempt_number              1
is_manual_scoring_required         False
learner_attempt_status      fully scored
is_deleted                         False

- No new values

###### Display counts of dates

In [12]:
date_min_max_null_unique(dfFlt, types['intervalVars'])

assignment_due_date
+------+----+-------------------+-------------------+
|unique|null|min                |max                |
+------+----+-------------------+-------------------+
|190   |1566|2019-08-23 04:59:00|2020-05-26 04:59:00|
+------+----+-------------------+-------------------+

assignment_final_submission_date
+------+----+-------------------+-------------------+
|unique|null|min                |max                |
+------+----+-------------------+-------------------+
|199   |1566|2019-08-23 04:59:00|2020-06-05 04:59:00|
+------+----+-------------------+-------------------+

assignment_start_date
+------+----+-------------------+-------------------+
|unique|null|min                |max                |
+------+----+-------------------+-------------------+
|206   |1566|2019-08-21 05:00:00|2020-05-19 05:00:00|
+------+----+-------------------+-------------------+

item_attempt_end_datetime_utc
+------+-----+-----------------------+-----------------------+
|unique|null |min  

###### Common observations with null was_submitted_datetime_actual

In [13]:
nullDf.filter(F.col('was_submitted_datetime_actual').isNull()).count()

13994

- 13994 in common

###### Common observations with 0 raw_score

In [14]:
nullDf.filter(F.col('raw_score') == 0).count()

11701

- 11701 in common

#### Conclusion
- address with the 15710 null assessment_item_response_id solution

In [5]:
finish_todo('Investigate 14999 null dates in item attempt dates')